vm/actor/src/builtin/miner/mod.rs

```diff
@@ -94,90 +120,127 @@ impl Actor {
         BS: BlockStore,
         RT: Runtime<BS>,
     {
-        rt.validate_immediate_caller_is(std::iter::once(&*INIT_ACTOR_ADDR))?;
+        rt.validate_immediate_caller_is(&[*INIT_ACTOR_ADDR])?;
 
         if !check_supported_proof_types(params.seal_proof_type) {
-            return Err(ActorError::new(
-                ExitCode::ErrIllegalArgument,
-                format!(
-                    "proof type {:?} not allowed for new miner actors",
-                    params.seal_proof_type
-                ),
+            return Err(actor_error!(
+                ErrIllegalArgument,
+                "proof type {:?} not allowed for new miner actors",
+                params.seal_proof_type
             ));
         }
 
-        let owner = resolve_owner_address(rt, params.owner)?;
+        let owner = resolve_control_address(rt, params.owner)?;
         let worker = resolve_worker_address(rt, params.worker)?;
+        let control_addresses: Vec<_> = params
+            .control_addresses
+            .into_iter()
+            .map(|address| resolve_control_address(rt, address))
+            .collect::<Result<_, _>>()?;
 
         let empty_map = make_map::<_, ()>(rt.store()).flush().map_err(|e| {
-            ActorError::new(
-                ExitCode::ErrIllegalState,
-                format!("failed to construct miner state: {}", e),
-            )
+            actor_error!(ErrIllegalState, "failed to construct initial state: {}", e)
         })?;
 
         let empty_array = Amt::<Cid, BS>::new(rt.store()).flush().map_err(|e| {
-            ActorError::new(
-                ExitCode::ErrIllegalState,
-                format!("failed to construct miner state: {}", e),
-            )
+            actor_error!(ErrIllegalState, "failed to construct initial state: {}", e)
         })?;
 
-        let empty_deadlines_cid = rt.store().put(&Deadlines::new(), Blake2b256).unwrap();
+        let empty_bitfield_cid = rt.store().put(&BitField::new(), Blake2b256).unwrap();
+
+        let empty_deadline_cid = rt
+            .store()
+            .put(&Deadline::new(empty_array.clone()), Blake2b256)
+            .unwrap();
+
+        let empty_deadlines_cid = rt
+            .store()
+            .put(&Deadlines::new(empty_deadline_cid), Blake2b256)
+            .unwrap();
+
+        let empty_vesting_funds_cid = rt.store().put(&VestingFunds::new(), Blake2b256).unwrap();
```

Comment:
These errors are still not downcasted, and no reason not too, since a direct interaction with the store. It's okay for the error to be incorrect until I make another pass, but I'd really like to avoid a potential crash when using.

Can probably downcast all of these errors at once if you put this logic in a smaller scope